Funcky/Extensions/ListExtensions.cs

```diff
@@ -7,4 +7,28 @@ public static class ListExtensions
     [Pure]
     public static Option<int> IndexOfOrNone<TValue>(this IList<TValue> list, TValue value)
         => MapNotFoundToNone(list.IndexOf(value));
+
+    [Pure]
+    public static Option<int> FindIndexOrNone<TValue>(this List<TValue> list, Predicate<TValue> match)
```

Comment:
A bit unfortunate that we have `Predicate<T>` here, but I guess its better to be consistent with `List`'s interface here?

Reply:
Otherwise we would have to Wrap the Func<T> in a Predicate<T>, that does not sound more elegant.